src/connection.ts

```diff
@@ -87,8 +88,21 @@ export class Connection {
     })
   }
 
-  public on(_event: "metadata_update", listener: MetadataUpdateListener) {
-    this.decoder.on("metadata_update", listener)
+  public on(
+    _event: "metadata_update" | "credit_response" | "deliver",
```

Comment:
the `_` is only for arguments not used so rename ti in `event`

Reply:
Why do you have also exposed also the `deliver` event? I suggest to keep it "private"